theano/tensor/scipy_ndimage.py

```diff
@@ -0,0 +1,47 @@
+import theano
+from theano.gradient import DisconnectedType
+from scipy.ndimage.interpolation import zoom
+import numpy as np
+
+class Zoom(theano.Op):
+    # Properties attribute
+    __props__ = ()
+
+    def __init__(self):
+        super(Zoom, self).__init__()
+
+    def make_node(self, x, ref):
+        x = theano.tensor.as_tensor_variable(x)
+        ref = theano.tensor.as_tensor_variable(ref)
+        return theano.Apply(self, [x, ref], [x.type()])
+
+    # Python implementation:
+    def perform(self, node, inputs, outputs_storage):
+        x, ref = inputs
+        z = outputs_storage[0]
+        assert x.ndim == ref.ndim
+        assert x.shape[0] == 1
+        #
+        resize_factors = [
+            ref.shape[2] / x.shape[2],
+            ref.shape[3] / x.shape[3]
```

Comment:
I think this might be the cause of some of your gradient problems. Your shapes are integers, so your `resize_factors` will be computed as integers as well. That doesn't work for the gradient of a `zoom(2)` upsampling, because you'd have to downsample the gradient with `zoom(0.5)`.

Reply:
I've added the `from __future__ import division` line which will ensure that using the `/` operator will return a floating point number. Tests confirm this behavior so this should be ok.